### ISSUE TITLE
Total active time calculation

### ISSUE THREAD

**Issue by jacobwhood** [SupremeTechnopriest/react-idle-timer#133]
Hello again!

Thanks for adding in the `getTotalActiveTime` function. We greatly appreciate it. I had a question about the calculation used to update this time as the user becomes idle. Correct me if I'm wrong but from what I understand, the user becomes idle once the idle timeout period has elapsed, after which the state is set to idle in `_toggleIdleState`. So the user is technically idle during the idle timeout period, correct? If this is the case, should the [active time update](https://github.com/SupremeTechnopriest/react-idle-timer/blob/46c39a45670950ae5f64f72cce8c77b7bcbe515a/src/useIdleTimer.js#L68) look something like this -- 
```
activeTime.current += (+new Date()) - timeout - lastActive.current
```

**Comment by SupremeTechnopriest** [comment]
Yeah you are correct.  Technically they are idle during the timeout period and it should probably be subtracted from the running active time total.  I will update the tests to account for this and publish a patch.  Good catch.

**Comment by SupremeTechnopriest** [comment]
I wrote some tests on this today, and I'm not really happy with the current implementation.  `getTotalActiveTime()` can return inaccurate values if it's called before an idle event.  I'm going to take some time to find a better way of implementing so it will always return an accurate value. I will take the `timeout` value into account.  Give me a few days.

**Comment by SupremeTechnopriest** [comment]
@jacobwhood Okay I am happy with this refactor.  Could you do me a favor and install directly from github and make sure it does what you expect?  I also added `getTotalIdleTime` and `getLastIdleTime` to make the API complete.  It's pretty throughly tested, but once you give me the thumbs up I will publish to npm.

**Comment by SupremeTechnopriest** [comment]
@jacobwhood Have you had some time to test this out?  I plan to publish it tomorrow.

**Comment by jacobwhood** [comment]
Hey man, I haven't had time -- just got moved onto a new project where we're super crunched for time. Not sure when I'll be moving back to what I was working on which depends on the timer. I'm going to try to get to it this weekend.

Thanks again for your responsiveness and quick work. 👍

**Comment by SupremeTechnopriest** [comment]
@jacobwhood No worries!  I will release this.  I think it looks pretty good.  Just reopen if you have any issues.  Good luck on the new project!

**Comment by SupremeTechnopriest** [comment]
Published as `4.5.0`.

**Comment by Floriferous** [comment]
Hi there, I'm a bit confused about this active time calculation. 

We'd like to store in our DB the total time a user spends on a certain page, so every 5-10 seconds, we'd like to send the additional time spent to our server, total active time seems like the perfect fit for this.

However, this is the value of `getTotalActiveTime()` in an interval of 5 seconds:

<img width="392" alt="Screenshot 2020-11-17 at 11 29 43" src="https://user-images.githubusercontent.com/10428082/99378846-33a5b480-28c8-11eb-98e4-31605d0dbc07.png">

I expect these values to be strictly continuously increasing, as I spend more time on this page, on and off. However, as you can see, the value jumps back and forth, most of the time going up, but sometimes going down.

Is this the expected behavior, or is something going wrong?
